### DISCUSSION TITLE
LTDC branch, program freezes when changing the resolution

### DISCUSSION THREAD

**Discussion by epikao** [stm32duino/Arduino_Core_STM32#1667] [Libraries]
Hello,

I have a question to the following branch / LTDC library part:
https://github.com/fpistm/Arduino_Core_STM32/tree/LTDC_TS_TEST_1.9.0/libraries

When I change 480 and 272 to another value as for example 1024 and 600 pixel, the program freezes immediately (directly after plugging in)
What is the best way to solve this? I am grateful for any hint.

Thank you

```
constexpr LTDCSettings LTDC_F746_ROKOTECH = {
    .width = 480,
    .height = 272,
    .horizontalSync = 41,
    .horizontalFrontPorch = 32,
    .horizontalBackPorch = 13,
    .verticalSync = 10,
    .verticalFrontPorch = 2,
    .verticalBackPorch = 10,
};

class LTDC_F746_Discovery : public LTDCClass<480, 272> {
    public:
    LTDC_F746_Discovery(): LTDCClass(LTDC_F746_ROKOTECH) {
```

**Comment by fpistm** [comment]
Hi @epikao 
it was shared as example. It is only a porting of example done by contributor.
Don't know why you want to change as the library is for the LCD of the Disco F746 which is 480*272.

Probably some other change are needed or configuration update (clock, buf size, ...).
Other example TS library also have to be updated (not exhaustive): https://github.com/fpistm/Arduino_Core_STM32/blob/b09706413a321d03a71469cbc34498b204a179d1/libraries/TS/src/stm32_ub_touch_480x272.h#L38-L39

**Reply by epikao** [reply]
> Don't know why you want to change as the library is for the LCD of the Disco F746 which is 480*272.

Thank you. I use the disco with another display (10.1") via adapter. I know that clock and timing and so on should also fit with the display, but the problems starts much earlier.
As soon as the resolution is changed (in the LTDC_F746 file), the program crashes right after plugging it in. I verify this by a LED blinking in the loop.
I'm afraid something with memory overflow or something. Would be great if you could look at that? 
I don't know memory/stack C/C++ so well/deep unfortunately...

The 10.1" display works well with 480x272 see attached image. The touch-function I have disabled for the moment.
![image](https://user-images.githubusercontent.com/26655950/156339282-260a03c4-c9c5-4629-92ab-33ae79bf1fec.png)


There is also a difference between the origin stm32-disco framebuffer definition and your definition, see below.
I remarked that the official stm32-disco code use SDRAM ... could that be an issue?

**stm32duino:**
`uint16_t *buffer = (uint16_t *)malloc(LTDC_F746_ROKOTECH.width * LTDC_F746_ROKOTECH.height * sizeof(uint16_t));`

**stm32-disco tft.c file:**
```
static __IO uintpixel_t * my_fb = (__IO uintpixel_t*) (0x60000000);
  ...
  BSP_SDRAM_Init();
  HAL_EnableFMCMemorySwapping();
  uint32_t i;
  for(i = 0; i < (screenWidth * screenHeight) ; i++)
  {
        buffer[i] = 0;
  }
```

**Reply by fpistm** [reply]
Unfortunately, I will not have time to instigate this.
As stated it has been provided as example. That's why it is on a branch and not on the main, sorry.
You could try to ask on the forum. Some topic talk about this.
